docs/sdk/java.md

```diff
@@ -364,6 +364,47 @@ PersistedTable<String, Integer> table = PersistedTable.of("my-table", String.cla
 PersistedAppendingBuffer<Integer> buffer = PersistedAppendingBuffer.of("my-buffer", Integer.class);
 {% endhighlight %}
 
+### State Expiration
+
+Persisted states may be configured to expire and be deleted after a specified duration.
+This is supported by all types of state:
+
+{% highlight java %}
+@Persisted
+PersistedValue<Integer> table = PersistedValue.of(
+    "my-value",
+    Integer.class,
+    Expiration.expireAfterWriting(Duration.ofHours(1)));
+
+@Persisted
+PersistedTable<String, Integer> table = PersistedTable.of(
+    "my-table",
+    String.class,
+    Integer.class,
+    Expiration.expireAfterWriting(Duration.ofMinutes(5)));
+
+@Persisted
+PersistedAppendingBuffer<Integer> buffer = PersistedAppendingBuffer.of(
+    "my-buffer",
+    Integer.class,
+    Expiration.expireAfterWriting(Duration.ofSeconds(30)));
+{% endhighlight %}
+
+There are two expiration modes supported:
+
+{% highlight java %}
+Expiration.expireAfterWriting(...)
+
+Expiration.expireAfterReadingOrWriting(...)
+{% endhighlight %}
+
+The above modes differ in when expiration timers for the state will be registered, either only on writes or on both reads and writes.
+
+These timers will also be persisted as managed state. This allows the timers to still be effectively running while the application is down.
```

Comment:
Imho this is an implemention detail, and is it true for RocksDB? I thought state is TTLed by background compaction.
My suggestion is to remove that paragraph, but let’s see what Seth thinks.

Reply:
Agreed, I put my suggestion on the other PR.

Reply:
I was using the term timers here generically, just to not leak too much implementation details about how the state backends actually persist registered timers.
But I agree, Seth's suggestion in the other PR looks good, will apply that and merge!